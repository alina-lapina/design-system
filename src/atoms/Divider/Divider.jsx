import React from 'react';
import PropTypes from 'prop-types';

const Divider = ({ className, dark, light }) => (
	<hr className={`divider-line ${dark ? 'dark' : ''}${light ? 'light' : ''} ${className}`} />
);

Divider.defaultProps = {
	className: '',
	dark: false,
	light: false,
};

Divider.propTypes = {
	className: PropTypes.string,
	dark: PropTypes.bool,
	light: PropTypes.bool,
};

export default Divider;
