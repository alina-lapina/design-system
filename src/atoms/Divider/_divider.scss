
.divider-line {
  display: block;
  height: 2px;
  margin: 10px 0;
  position: relative;
  width: 100%;

  &.dark {
    border-top: 1px solid $ssb-dark-6;
  }

  &.light {
    border-top: 1px solid rgba($ssb-dark-6, .25);
  }
}