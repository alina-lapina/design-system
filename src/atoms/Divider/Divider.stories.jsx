import React from 'react';
import { storiesOf } from '@storybook/react';
import Divider from './Divider';

storiesOf('Atoms|Divider', module)
	.add('Dark', () => (
		<div style={{ alignItems: 'center', display: 'flex', height: '100%', position: 'absolute', width: '100%' }}>
			<Divider dark />
		</div>
	))
	.add('Light', () => (
		<div style={{ alignItems: 'center', display: 'flex', height: '100%', position: 'absolute', width: '100%' }}>
			<Divider light />
		</div>
	));
