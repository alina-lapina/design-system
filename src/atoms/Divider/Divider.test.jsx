import React from 'react';
import { shallow, mount } from 'enzyme';
import Divider from './Divider';

describe('Divider component', () => {
	test('Matches the snapshot', () => {
		const wrapper = shallow(<Divider dark />);
		expect(wrapper).toMatchSnapshot();
	});
	test('Decides className from props', () => {
		const lightWrapper = shallow(<Divider light />);
		const darkWrapper = shallow(<Divider dark />);
		expect(lightWrapper.find('hr').hasClass('light')).toEqual(true);
		expect(darkWrapper.find('hr').hasClass('dark')).toEqual(true);
	})
});
