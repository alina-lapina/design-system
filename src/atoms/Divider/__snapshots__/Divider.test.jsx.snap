// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Divider component Matches the snapshot 1`] = `
<hr
  className="divider-line dark "
/>
`;
